components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -114,7 +129,46 @@ void reconcileItemChecksum(final Repository repo, final StorageFileItem item) th
         finally {
           uid.getLock().unlock();
         }
+
+        final String itemPath = item.getPath();
+        if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
```

Comment:
can we just put this check right up front and skip the digest check on the checksum files themselves?